src/main/java/com/puppycrawl/tools/checkstyle/utils/JavadocUtils.java

```diff
@@ -420,4 +525,18 @@ public static boolean isCorrectJavadocPosition(DetailAST blockComment) {
                 || BlockCommentPosition.isOnEnumConstant(blockComment)
                 || BlockCommentPosition.isOnAnnotationDef(blockComment);
     }
+
+    /**
+     * When all the html tags are properly closed the resultant javadoc tree is nested. This method
+     * fetches the type of the parent as described by {@link JavadocTokenTypes} of a html tag in
+     * a nested javadoc tree.
+     * @param tagOpen the html tag for which the parent type is to be obtained assuming
+     *                proper nesting
+     * @return the type of the parent when the supplied tag results in proper nesting
+     * @see #NESTED_HTML_CHILD_PARENT_MAPPINGS
+     * @see com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocNodeImpl#hasUnclosedTag
+     */
+    public static int getParentTypeWhenNested(DetailNode tagOpen) {
```

Comment:
No. 3810 is only to allow javadoc trees to be nested when there is `SINGLETON_ELEMENT` in tags. For all tags to be nested all tags must be closed, which can't be ensured of course